staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -93,6 +93,10 @@ func NewValidator(s *schema.Structural, isResourceRoot bool, perCallLimit uint64
 // exist. declType is expected to be a CEL DeclType corresponding to the structural schema.
 // perCallLimit was added for testing purpose only. Callers should always use const PerCallLimit from k8s.io/apiserver/pkg/apis/cel/config.go as input.
 func validator(validationSchema, nodeSchema *schema.Structural, isResourceRoot bool, declType *cel.DeclType, perCallLimit uint64) *Validator {
+	if declType == nil {
+		klog.Errorf("Failed to convert to declType for CEL validation rules")
```

Comment:
if declType is nil, is returning nil correct, or should we be returning a validator with a non-nil compilationErr?

Reply:
If calling Compile return an error, an nil validator will be returned. So it would be the same behavior :)

Reply:
Updated to return validator with a non-nil compilationErr